plugins/CuraEngineBackend/CuraEngineBackend.py

```diff
@@ -301,41 +307,58 @@ def _onStartSliceCompleted(self, job):
 
         Logger.log("d", "Sending slice message took %s seconds", time() - self._slice_start_time )
 
+    ##  Determine enable or disable auto slicing. Return True for enable timer and False otherwise.
+    #   It disables when
+    #   - preference auto slice is off
+    #   - decorator isBlockSlicing is found (used in g-code reader)
+    def determineAutoSlicing(self):
+        enable_timer = True
+
+        if not Preferences.getInstance().getValue("general/auto_slice"):
+            enable_timer = False
+        for node in DepthFirstIterator(self._scene.getRoot()):
+            if node.callDecoration("isBlockSlicing"):
+                enable_timer = False
+                self.backendStateChange.emit(BackendState.Disabled)
+            gcode_list = node.callDecoration("getGCodeList")
+            if gcode_list is not None:
+                self._scene.gcode_list = gcode_list
+
+        if self._use_timer == enable_timer:
+            return
+        if enable_timer:
+            self.backendStateChange.emit(BackendState.NotStarted)
+            self.enableTimer()
+            return True
+        else:
+            self.disableTimer()
+            return False
+
     ##  Listener for when the scene has changed.
     #
     #   This should start a slice if the scene is now ready to slice.
     #
     #   \param source The scene node that was changed.
     def _onSceneChanged(self, source):
+        if self._tool_active:
+            return
+
         if type(source) is not SceneNode:
             return
 
         if source is self._scene.getRoot():
             return
 
-        should_pause = self._pause_slicing
-        block_slicing = False
-        for node in DepthFirstIterator(self._scene.getRoot()):
-            if node.callDecoration("isBlockSlicing"):
-                should_pause = True
-                block_slicing = True
-            gcode_list = node.callDecoration("getGCodeList")
-            if gcode_list is not None:
-                self._scene.gcode_list = gcode_list
-
-        self._block_slicing = block_slicing
-
-        if should_pause or self._block_slicing:
-            self.pauseSlicing()
-        else:
-            self.continueSlicing()
+        self.determineAutoSlicing()
```

Comment:
Shouldn't we do something with the result of the determineAutoSlicing function here?